src/Stryker.Core/Stryker.Core.UnitTest/Initialisation/InitialisationProcessTests.cs

```diff
@@ -67,6 +66,60 @@ public void InitialisationProcess_ShouldCallNeededResolvers()
 
             inputFileResolverMock.Verify(x => x.ResolveInput(It.IsAny<StrykerOptions>()), Times.Once);
         }
+        
+        [Fact]
+        public void InitialisationProcess_UpdatesBasePathToTestProjectLocation()
+        {
+            var testRunnerMock = new Mock<ITestRunner>(MockBehavior.Strict);
+            var inputFileResolverMock = new Mock<IInputFileResolver>(MockBehavior.Strict);
+            var initialBuildProcessMock = new Mock<IInitialBuildProcess>(MockBehavior.Strict);
+            var initialTestProcessMock = new Mock<IInitialTestProcess>(MockBehavior.Strict);
+            var assemblyReferenceResolverMock = new Mock<IAssemblyReferenceResolver>(MockBehavior.Strict);
+            
+            testRunnerMock.Setup(x => x.RunAll(It.IsAny<int>(), It.IsAny<int>()))
+                .Returns(new TestRunResult { Success = true }); // testrun is successful
+            inputFileResolverMock.Setup(x => x.ResolveInput(It.IsAny<StrykerOptions>()))
+                .Returns(new ProjectInfo
+                {
+                    ProjectUnderTestAnalyzerResult = new ProjectAnalyzerResult(null, null)
+                    {
+                        References = new string[0]
+                    },
+                    TestProjectAnalyzerResult = new ProjectAnalyzerResult(null, null)
+                    {
+                        ProjectFilePath = "C://Example/Dir/ProjectFolder/TestFolder/TestProject.csproj"
+                    },
+                    ProjectContents = new FolderComposite
+                    {
+                        Name = "ProjectRoot",
+                        Children = new Collection<ProjectComponent>
+                        {
+                            new FileLeaf
+                            {
+                                Name = "SomeFile.cs"
+                            }
+                        }
+                    }
+                });
+            initialTestProcessMock.Setup(x => x.InitialTest(It.IsAny<ITestRunner>())).Returns(999);
+            initialBuildProcessMock.Setup(x => x.InitialBuild(It.IsAny<bool>(), It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()));
+            assemblyReferenceResolverMock.Setup(x => x.LoadProjectReferences(It.IsAny<string[]>()))
+                .Returns(Enumerable.Empty<PortableExecutableReference>());
+
+            var target = new InitialisationProcess(
+                inputFileResolverMock.Object,
+                initialBuildProcessMock.Object,
+                initialTestProcessMock.Object,
+                testRunnerMock.Object,
+                assemblyReferenceResolverMock.Object);
+
+            var options = new StrykerOptions();
+
+            var result = target.Initialize(options);
+
+            inputFileResolverMock.Verify(x => x.ResolveInput(It.IsAny<StrykerOptions>()), Times.Once);
+            options.BasePath.ShouldBe( "C:/Example/Dir/ProjectFolder/TestFolder"); //ToDo: expect OS specific path
```

Comment:
You can use the PathUtils for the todo job :+1:

Reply:
I couldn't really make it work with PathUtils.
my input path was C://one/two/three
but the path detected by the system. Path.PathDirectory 
would be C:/one/two/three
missing one /

I'll try again (after moving this to the CLI)